### PR TITLE
memo: frontend report september 2024

### PR DESCRIPTION
**Pull request by ngolapnguyen**

### CODE REVIEW

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Lazy loading with React Router can be tricky. Avoid common pitfalls by understanding how to structure your loader functions for optimal performance.
+
+### Quick Links
+
+- [The unspoken rules of React hooks](https://macwright.com/2024/09/19/the-extra-rules-of-hooks)
+- [React Testing Technique: Inverse Assertions](https://www.epicweb.dev/inverse-assertions)
+- [Modern Data-Fetching in React - Comparing architectures at a high level](https://reacttraining.com/blog/modern-data-fetching-in-react)
+- [Vim for React Developers - A free course focusing on real-world scenarios](https://vimforreactdevs.com/)
+
+## Next.js
```
Comment (zlatanpham) [inline 01_literature/Engineering/Frontend/frontend-report-september-2024.md:34]: Seems there’s an important piece missing this month from https://www.reddit.com/r/nextjs/comments/1f92jdv/chatgptcom_switched_from_nextjs_to_remix/

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+### [Meet Conform: The Type-Safe Form Validation Library](https://conform.guide/)
+
+Conform is a type-safe form validation library utilizing web fundamentals to progressively enhance HTML Forms with full support for server frameworks like [Remix](https://remix.run/) and [Next.js](https://nextjs.org/).
+
+### [NewDevTools: Your Daily Dose of Developer Goodies](https://newdevtools.com/)
+
+Discover new developer tools & services daily with NewDevTools. Updated daily!
+
+### [Bulletproof React - A simple, scalable, and powerful architecture for building production ready React applications](https://github.com/alan2207/bulletproof-react)
```
Comment (zlatanpham) [inline 01_literature/Engineering/Frontend/frontend-report-september-2024.md:103]: 27k stars in one month? I’m not certain if it’s worth the attention that crazy 🙏

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+- [Write and Code with ChatGPT: Introducing Canvas](https://openai.com/index/introducing-canvas/)
+
+## Commentary
+
+- [I am tired of AI](https://www.ontestautomation.com/i-am-tired-of-ai/)
+- [Legacy Modernization meets GenAI](https://martinfowler.com/articles/legacy-modernization-gen-ai.html)
+- [Why Copilot is Making Programmers Worse at Programming](https://www.darrenhorrocks.co.uk/why-copilot-making-programmers-worse-at-programming/)
+- [Why Gumroad Didn't Choose htmx](https://htmx.org/essays/why-gumroad-didnt-choose-htmx/)
+- [The hardest bug investigation of my career and the insane code that caused it](https://www.reddit.com/r/ExperiencedDevs/comments/1fu0e5q/the_hardest_bug_investigation_of_my_career_and/)
+- [Why Not Comments - Why not "why not" comments? Not why "not comments"](https://buttondown.com/hillelwayne/archive/why-not-comments/)
```
Comment (zlatanpham) [inline 01_literature/Engineering/Frontend/frontend-report-september-2024.md:122]: IMO, only the htmx link is related to the frontend

**Review by zlatanpham — COMMENTED** [review]
I’ve left some comments. A couple of them relate to frontend relevancy

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md (hunk 1 above)
Comment (ngolapnguyen) [reply 01_literature/Engineering/Frontend/frontend-report-september-2024.md:34]: My bad, somehow I remembered it's from Aug

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+- [The web's clipboard, and how it stores data of different types](https://alexharri.com/blog/clipboard)
+- [CSS display contents - How it works](https://ishadeed.com/article/display-contents/)
+- [Cleaner JavaScript promises with safe-await](https://alexpate.com/posts/cleaner-promises-with-safe-await)
+- [CSS Masonry & CSS Grid](https://css-tricks.com/css-masonry-css-grid/)
+
+## Trending
+
+### [The Intelligence Age](https://ia.samaltman.com/)
+
+Sam Altman talks about the exciting future of AI and how it's set to revolutionize everything from healthcare to fighting climate change.
```
[the remark this answers is not in the document]
Comment (ngolapnguyen) [reply 01_literature/Engineering/Frontend/frontend-report-september-2024.md:78]: It's just trending + might be meaningful to all aspects of engineering, not just frontend

[the remark this answers is not in the document]
Comment (ngolapnguyen) [reply 01_literature/Engineering/Frontend/frontend-report-september-2024.md:78]: I think I'll move this down to commentary, actually

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md (hunk 2 above)
Comment (ngolapnguyen) [reply 01_literature/Engineering/Frontend/frontend-report-september-2024.md:103]: I meant, it's included in many newsletters, so there's certainly a trend

File: 01_literature/Engineering/Frontend/frontend-report-september-2024.md (hunk 3 above)
Comment (ngolapnguyen) [reply 01_literature/Engineering/Frontend/frontend-report-september-2024.md:122]: Some I find are worth reading for its engineering aspect. These are just commentaries anyway 🤔

**Review by zlatanpham — APPROVED** [review]

**Pull request merged**